addons/api/addon/services/resource-filter-store.js

```diff
@@ -0,0 +1,109 @@
+import Service from '@ember/service';
+import { inject as service } from '@ember/service';
+
+/**
+ * Takes a POJO representing a query and converts it to a boolean expression
+ * string which is understood by the API.
+ *
+ * @example
+ *   const filter = new ResourceFilter({
+ *     type: 'oidc',
+ *     status: ['active', 'pending']
+ *   });
+ *   console.log(filter.queryExpression);
+ *   // ("/item/type" == "oidc") and ("/item/status" == "active" or "/item/status" == "pending")
+ */
+export class ResourceFilter {
+  // =attributes
+
+  /**
+   * @type {?object}
+   */
+  #filterObject;
+
+  /**
+   * @type {string}
+   */
+  get queryExpression() {
+    const clauses = Object.keys(this.#filterObject)
+      .map((key) => {
+        const value = this.#filterObject[key];
+        const valuesArray = Array.isArray(value) ? value : [value];
+        return this.parenthetical(
+          this.or(valuesArray.map((value) => this.equals(key, value)))
+        );
+      })
+      .flat();
+    return this.and(clauses);
+  }
+
+  // =methods
+
+  /**
+   * @param {object} filterObject
+   */
+  constructor(filterObject = {}) {
+    this.#filterObject = Object.assign({}, filterObject);
+  }
+
+  /**
+   * @param {string} key
+   * @param value
+   * @return {string}
+   */
+  equals(key, value) {
+    return `"/item/${key}" == "${value}"`;
+  }
+
+  /**
+   * @param {string[]} clauses
+   * @return {string}
+   */
+  and(clauses) {
+    return clauses.join(' and ');
+  }
+
+  /**
+   * @param {string[]} clauses
+   * @return {string}
+   */
+  or(clauses) {
+    return clauses.join(' or ');
+  }
+
+  /**
+   * @param {string} expression
+   * @return {string}
+   */
+  parenthetical(expression) {
+    return `(${expression})`;
+  }
+}
+
+/**
+ *
+ */
+export default class ResourceFilterStoreService extends Service {
+  // =services
+
+  @service store;
+
+  // =methods
+
+  /**
+   * Similar to Ember's built-in `store.query`, except this method accepts
+   * an additional argument `filterObject` representing a query filter
+   * (see ResourceFilter class above).
+   * @param {string} modelName
+   * @param {object} filterObject
+   * @param {?object} storeQuery
```

Comment:
is `?object` a typo?

Reply:
Intentional.  This indicates that the argument is optional.  Technically both objects are optional, but the whole point of this service is to pass a filter object.